Couchsurfing/rails-challenge
# luis-metzger-code-challenge-submission

## Tester Notes
- bundle
 - Run migrations
 - Create customer, product, and variants. (if you have not already)
 - Run rails server

[POST] Create Endpoint: `http://localhost:3000/api/v1/create_order`

`Create Test POST data: 

{ 
	"customer_id": 1,
	"variant_ids": [
		 				{ 
		 					"variant_id": 1, 
		 				    "variant_quantity": 1
		 				}, 
		 				{ 
		 					"variant_id": 2,
		 					"variant_quantity": 1
		 					
		 				}, 
		 				{ "variant_id": 3, 
		 				  "variant_quantity":  1
		 					
		 				}
		 			]
}`


[GET] Show Endpoint: http://localhost:3000/api/v1/customer_orders/:id 

`Expected data: {
    "id": 158,
    "total_cost": 55,
    "order_status": "pending",
    "created_at": "2019-06-25T21:25:33.537Z",
    "updated_at": "2019-06-25T21:25:33.845Z",
    "variant_orders": [
        {
            "variant_id": 1,
            "variant_quantity": 1,
            "variant": {
                "id": 1,
                "name": "Large",
                "cost": 25
            }
        },
        {
            "variant_id": 2,
            "variant_quantity": 1,
            "variant": {
                "id": 2,
                "name": "Medium",
                "cost": 20
            }
        },
        {
            "variant_id": 3,
            "variant_quantity": 1,
            "variant": {
                "id": 3,
                "name": "Small",
                "cost": 10
            }
        }
    ]
}`
